server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Id.scala

```diff
@@ -0,0 +1,35 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.model
+
+import eu.timepit.refined.api.{Refined, Validate}
+
+object Id {
+  implicit val idTypeValidate: Validate.Plain[String, IdType] =
```

Comment:
I cannot using the existing 
```
  type Id = String Refined And[
    Size[Interval.Closed[1, 255]],
    MatchesRegex["^[a-zA-Z0-9-_]*$"]]
```

it breaks `refinveV()` - the method is recommended to be using in runtime. 

Literal values: `Id("abc-xyz")`
Non Literal values: 
```
var abc: String = ...
val either = refineV[Id](abc)

either match {
   case Left(failure) =>
   case Right(id) =>
}
```